src/sonic-eventd/rsyslog_plugin/rsyslog_plugin.cpp

```diff
@@ -37,14 +45,17 @@ bool RsyslogPlugin::createRegexList() {
         return false;
     }
 
-    string regexString;
     regex expression;
 
     for(long unsigned int i = 0; i < m_parser->m_regexList.size(); i++) {
+        string regexString = "([a-zA-Z]{3} [0-9]{1,2} [0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{0,6}) ";
```

Comment:
Is not the timestamp is the fixed length string from the start of the log message. Why not just get substr(0, 22)

Reply:
Applying regex, not sure about the efficiency of creating a substring of 25 characters than applying the regex versus applying the regex on the whole string from the start

Reply:
guess you are correct. Add carrot at the start, so it can only match at the start of the string